include/openthread/platform/radio.h

```diff
@@ -38,6 +38,7 @@
 
 #include <stdint.h>
 
+#include <openthread/config.h>
```

Comment:
Be core config?

Reply:
The `openthread-core-config.h` should not be included by the headers in `include`. OPENTHREAD_CONFIG_RADIO_2P4GHZ_OQPSK_SUPPORT is enabled in `openthread-core-config.h`by default and the parameters in`radio.h` is set to support RADIO_2P4GHZ_OQPSK by default. If user want to modify the modulation definition, he should add the new definition to the OPENTHREAD_CONFIG_FILE.

Reply:
Correct. We cannot and **should not** include "openthread-core-config.h" in any public header file of OpenThread Note that "openthread-core-config.h" is not a public header file. Furthermore we let users to specific their project specific configuration header file  (we cannot make a project specified header file public from OT library as OT lib does not even own such a file). 
Related to "openthread-config.h", note that this is generated by autoconf so we cannot expect users to go and edit it.  Again we should try to avoid including this in any of OT public header files. 

This all means, we cannot use `#if (OPENTHREAD_CONFIG_RADIO_2P4GHZ_OQPSK_SUPPORT && OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT)` in this header file or any public header file. 

Here is a possible solution I can suggest to make it work in this case: 

- We can change the `OT_RADIO_NUM_PAGES`, `OT_RADIO_SUPPORTED_CHANNEL` , `OT_RADIO_CHANNEL_MIN`, ... from an `enum` to be declared as `extern const` variables in this header file.
```
extern const uint8_t  OT_RADIO_NUM_PAGES;
extern const uint32_t OT_RADIO_SUPPORTED_CHANNELS;
...
```
- We can then define these variable in a core file (say `sub_mac.cpp`) where we can use the `#if OPENTHREAD_CONFIG_RADIO_2P4GHZ_OQPSK_SUPPORT` and have access to config options

Reply:
You are right, `#if (OPENTHREAD_CONFIG_RADIO_2P4GHZ_OQPSK_SUPPORT && OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT)` should not be put in `radio.h`. 
- This method is not common, so far I haven't seen such a usage in OT's header files. Besides, these constants looks like macros，but they are actually  const variable. If user use them to define arrays, it will generate compile errors. These constants are physical layer parameters. They may not be appropriate to be put in file `sub_mac.cpp` .

I think it is better to add two new files `src/core/phy/phy.cpp` and `src/core/phy/phy.hpp`. Then put these constants in `src/core/phy/phy.hpp` and provide APIs to get some these constants in file `src/core/phy/phy.cpp`.

Comments?